pkg/chain/ethereum/balance_monitor.go

```diff
@@ -0,0 +1,70 @@
+package ethereum
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/keep-network/keep-core/pkg/chain"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// BalanceSource provides a balance info for the given address.
+type BalanceSource func(address common.Address) (*big.Int, error)
```

Comment:
Do we need this function given that this code is located in `ethereum` package? Why not to call `WeiBalanceOf` directly (or `BalanceAt` per another comment)? I think we are adding another level of indirection here that is not really needed.

Reply:
This code is also used by `keep-ecdsa` so this is why we need an indirection here.

Reply:
Ah, I see. And `keep-ecdsa` has its own `ethereum` struct that is not referring to this one.